src/TriggerService/TriggerHostedService.cs

```diff
@@ -508,11 +508,24 @@ private async Task UploadTimingAsync(Guid id, string sampleName)
         private async Task<WorkflowFailureInfo> GetWorkflowFailureInfoAsync(Guid workflowId, string metadata)
         {
             const string BatchExecutionDirectoryName = "__batch";
+            const int maxFailureMessageLength = 4096;
 
             var metadataFailures = string.IsNullOrWhiteSpace(metadata)
                 ? default
                 : JsonConvert.DeserializeObject<CromwellMetadata>(metadata)?.Failures;
 
+            if (metadataFailures?.Count > 0)
+            {
+                // Truncate failure messages; some can be 56k+ when it's an HTML message
+                for (int i = 0; i < metadataFailures.Count; i++)
+                {
+                    if (metadataFailures[i].Message?.Length > maxFailureMessageLength)
+                    {
+                        metadataFailures[i].Message = metadataFailures[i].Message[..maxFailureMessageLength];
```

Comment:
I would suggest adding some marker that the message has been truncated, e.g. elipsis or "[Truncated]".

Reply:
Great idea